openc3/lib/openc3/models/plugin_model.rb

```diff
@@ -267,7 +267,14 @@ def as_json(*a)
 
     # Undeploy all models associated with this plugin
     def undeploy
-      [ToolModel, TargetModel, InterfaceModel, RouterModel, MicroserviceModel, WidgetModel].each do |model|
+      MicroserviceModel.find_all_by_plugin(plugin: @name, scope: @scope).each do |name, model_instance|
+        model_instance.destroy
+      end
+      # Wait for the operator to wake up and remove the microservice processes
+      sleep 30 # Cycle time 15s plus 2s wait for soft stop and then hard stop
```

Comment:
I didn't really know the best way to signal the MicroserviceOperator that it's time to remove processes so I just left the sleep. If you want to optimize from here that would be awesome.

Reply:
I'm going to decrease the operator cycle time to 5s and reduce this.